File: backend/app/models/portfolio.py
from typing import Optional

from pydantic import BaseModel


class Position(BaseModel):
    code: str
    stock_name: str
    market: str
    qty: float
    cost_price: float
    current_price: float
    market_val: float
    pl_val: float
    pl_ratio: float
    day_change: float
    day_change_ratio: float
    currency: str
    sector: Optional[str] = None


class PortfolioSummary(BaseModel):
    total_market_value: float
    total_cost: float
    total_pl_val: float
    total_pl_ratio: float
    total_day_change: float
    position_count: int
    currency_breakdown: dict[str, float]
    diversification_score: int
    top_holding_concentration: float


class SectorBreakdown(BaseModel):
    sector: str
    weight_pct: float
    market_value: float
    positions: list[str]

File: frontend/src/types/portfolio.ts
export interface Position {
  code: string
  stock_name: string
  market: string
  qty: number
  cost_price: number
  current_price: number
  market_val: number
  pl_val: number
  pl_ratio: number
  day_change: number
  day_change_ratio: number
  currency: string
  sector: string | null
}

export interface PortfolioSummary {
  total_market_value: number
  total_cost: number
  total_pl_val: number
  total_pl_ratio: number
  total_day_change: number
  position_count: number
  currency_breakdown: Record<string, number>
  diversification_score: number
  top_holding_concentration: number
}

export interface SectorBreakdown {
  sector: string
  weight_pct: number
  market_value: number
  positions: string[]
}
